server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreStatsFromNodesStatsIT.java

```diff
@@ -0,0 +1,210 @@
+            .getRemoteTranslogStats();
+        assertTrue(remoteTranslogStats.getUploadBytesStarted() > 0);
+        assertTrue(remoteTranslogStats.getUploadBytesSucceeded() > 0);
+    }
+
+    private void assertZeroRemoteSegmentStats(RemoteSegmentStats remoteSegmentStats) {
+        // Compare with fresh object because all values default to 0 in default fresh object
+        assertEquals(new RemoteSegmentStats(), remoteSegmentStats);
+    }
+
+    private void assertZeroRemoteTranslogStats(RemoteTranslogStats remoteTranslogStats) {
+        // Compare with fresh object because all values default to 0 in default fresh object
+        assertEquals(new RemoteTranslogStats(), remoteTranslogStats);
+    }
+
+    private static void assertNodeStatsParityAcrossNodes(String... indices) {
+        for (String dataNode : internalCluster().getDataNodeNames()) {
+            assertNodeStatsParityOnNode(dataNode, indices);
+        }
+    }
+
+    private static void assertNodeStatsParityOnNode(String dataNode, String... indices) {
+        RemoteSegmentStats remoteSegmentStatsCumulative = new RemoteSegmentStats();
+        RemoteTranslogStats remoteTranslogStatsCumulative = new RemoteTranslogStats();
+        for (String index : indices) {
+            // Fetch _remotestore/stats
+            RemoteStoreStatsResponse remoteStoreStats = client(dataNode).admin()
+                .cluster()
+                .prepareRemoteStoreStats(index, "0")
+                .setLocal(true)
+                .get();
+            remoteSegmentStatsCumulative.add(new RemoteSegmentStats(remoteStoreStats.getRemoteStoreStats()[0].getSegmentStats()));
+            remoteTranslogStatsCumulative.add(new RemoteTranslogStats(remoteStoreStats.getRemoteStoreStats()[0].getTranslogStats()));
+        }
+
+        // Fetch _nodes/stats
+        NodesStatsResponse nodesStatsResponse = client().admin()
+            .cluster()
+            .prepareNodesStats(dataNode)
+            .setIndices(new CommonStatsFlags().set(CommonStatsFlags.Flag.Segments, true).set(CommonStatsFlags.Flag.Translog, true))
+            .get();
+
+        // assert segment stats
+        RemoteSegmentStats remoteSegmentStatsFromNodesStats = nodesStatsResponse.getNodes()
+            .get(0)
+            .getIndices()
+            .getSegments()
+            .getRemoteSegmentStats();
+        assertEquals(remoteSegmentStatsCumulative, remoteSegmentStatsFromNodesStats);
+        // Ensure that total upload time has non-zero value if there has been segments uploaded from the node
+        if (remoteSegmentStatsCumulative.getUploadBytesStarted() > 0) {
+            assertTrue(remoteSegmentStatsCumulative.getTotalUploadTime() > 0);
+        }
+        // Ensure that total download time has non-zero value if there has been segments downloaded to the node
+        if (remoteSegmentStatsCumulative.getDownloadBytesStarted() > 0) {
+            assertTrue(remoteSegmentStatsCumulative.getTotalDownloadTime() > 0);
+        }
```

Comment:
Thanks for simplifying the code.

In the earlier code, we are asserting that the cumulative upload started, succeeded and time is greater than 0 for segments case. Right now, we have made this conditional. Is there any way we have those `>0` assertions run always based on a arg maybe? I understand that we have tried to reuse same method for upload and download both.

Reply:
>Right now, we have made this conditional.

In the first segment stats test, this wasn't checked at all. We were only comparing the cumulative number with the nodes stats number.

>  Is there any way we have those >0 assertions run always based on a arg maybe? I understand that we have tried to reuse same method for upload and download both.

Will check. But is this a blocker for the PR merge? I have fixed some code blocks which weren't readable. I wouldn't want to risk the PR merge for 2.10 release due to refactoring changes.

Reply:
@ashking94 the change we discussed over call to also assert for `> 0` values will need much more refactoring as the assertion also depends on whether the current node had any primary shards or not. Additionally, I think we would be duplicating some logic from `RemoteStoreStatsIT` if we were to add these assertions, while this IT suite is focused on parity between the two APIs in context.

Please let me know if we still need the additional assertions. We can track the refactoring as a separate GH issue and release this PR for 2.10.